src/models/task/impls.rs

```diff
@@ -82,6 +88,47 @@ impl Ord for Task {
 }
 
 impl Task {
+    /// Create new task
+    /// ## Parmeters:
+    /// - task_id: id of the task
+    /// - title: title of the task
+    /// - duration: duration of the task
+    /// - goal: goal of the task
+    /// - timeline: timeline of the task
+    /// - status: status of the task
+    /// - timeframe: Start and deadline of a task
+    pub fn new(
+        task_id: usize,
+        title: &str,
+        duration: usize,
+        goal: &Goal,
+        timeline: &Timeline,
+        status: &TaskStatus,
+        timeframe: Option<Slot>,
```

Comment:
@moaz-mokhtar I don't see why this is required when you have the `timeline`.  
If it's temporary for refactoring please indicate this in code.

Reply:
I did create this function for below:
- the challenge to avoid splitting slots outside `Timeline`, which I found 8 times slots splitting outside Timeline; 2 of them are repeated. So I'm extract functions outside calls inside iterator or other functionalities which splitting slots.
- as agreed before that create new task should be from a function not inside `Budget` or other staff; so I found ~ 4 calls for creating new `Task` and I just created a function to be common. It can be better

Reply:
Ok I understand now it's about standardizing before doing further refactoring. Perfect!